include/dr/mhp/halo.hpp

```diff
@@ -363,11 +363,11 @@ class span_halo : public span_halo_impl<T, Memory> {
     drlog.debug(nostd::source_location::current(),
                 "owned groups {}/{} first/last\n", comm.first(), comm.last());
     if (hb.prev > 0 && (hb.periodic || !comm.first())) {
```

Comment:
```suggestion
    if (hb.next > 0 && (hb.periodic || !comm.first())) {
```

If next halo is > 0, then we send to previous to updates its 'next' halo

Reply:
I understand this suggested change fixes some bug. @mateuszpn , could you please add test cases which fails before and passes after applying this change?